homeassistant/components/devolo_home_network/__init__.py

```diff
@@ -39,6 +39,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         device = Device(
             ip=entry.data[CONF_IP_ADDRESS], zeroconf_instance=zeroconf_instance
         )
+        device.password = entry.data[CONF_PASSWORD]
```

Comment:
How does this work for users upgrading from a previous version where there's no password in the config entry data?

Reply:
hmm, I indeed did not think about that case. Is it ok to switch to get?

Reply:
Yes, it's fine to use `get`, please add a comment explaining the key was added in HA Core 2022.6
Please also add a test where the config entry data lacks the password to test the behavior.

Reply:
I added both.